packages/eslint-plugin/src/rules/prefer-literal-enum-member.ts

```diff
@@ -0,0 +1,41 @@
+import { createRule } from '../util';
+import { AST_NODE_TYPES } from '@typescript-eslint/experimental-utils';
+import { TSEnumMember } from '../../../typescript-estree/dist/ts-estree/ts-estree';
+
+export default createRule<[], 'notLiteral'>({
+  name: 'prefer-literal-enum-member',
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description:
+        'Require that all enum members be literal values to prevent unintended enum member name shadow issues',
+      category: 'Best Practices',
+      recommended: false,
+      requiresTypeChecking: false,
+    },
+    messages: {
+      notLiteral: `Enum member must be a ${AST_NODE_TYPES.Literal}.`,
+    },
+    schema: [],
+  },
+  defaultOptions: [],
+  create(context) {
+    return {
+      TSEnumMember(node: TSEnumMember): void {
+        // If there is no initializer, then this node is just the name of the member, so ignore.
+        if (
+          node.initializer != null &&
+          node.initializer.type !== AST_NODE_TYPES.Literal
```

Comment:
You can use a selector to do this

```suggestion
      'TSEnumMember[initializer != null]'(node: TSEnumMember): void {
        if (
          node.initializer?.type !== AST_NODE_TYPES.Literal
```

Reply:
I can do this, but I'm not sure how useful it is.  In fact, is seems that the nullish check will now happen twice at runtime due to the addition of the optional chaining operator.  Not a huge performance penalty, I'm sure.  Perhaps I could just use the optional chaining operator and skip the selector?

Reply:
So, the Node 10 unit tests failed.  I went with the selector and a non-null assertion in the code.  We'll see how that goes.

Reply:
Unit tests also fail for Node 10 when using the selector.  I'm reverting that and just doing an explicit nullish check in the code.